src/test/java/com/study/realworld/user/service/UserServiceTest.java

```diff
@@ -161,54 +158,76 @@ void save() {
     }
 
     @Test
-    void deleteById() {
+    void login() {
+        // given
+        final UserLoginRequest request = UserLoginRequest.builder()
+                .email(EMAIL)
+                .password(PASSWORD)
+                .build();
 
         final User user = User.builder()
                 .email(EMAIL)
                 .username(USERNAME)
-                .password(passwordEncoder.encode(PASSWORD))
+                .password(PASSWORD)
                 .bio(BIO)
                 .image(IMAGE)
                 .build();
 
-        when(userRepository.findById(anyLong()))
+        when(userRepository.findByEmail(anyString()))
                 .thenReturn(Optional.ofNullable(user));
 
-        //doNothing().when(userRepository).delete(any());
+        /**
+         * This exception may occur if matchers are combined with raw values:
+         *     //incorrect:
+         *     someMethod(anyObject(), "raw String");
+         * When using matchers, all arguments have to be provided by matchers.
+         * For example:
+         *     //correct:
+         *     someMethod(anyObject(), eq("String by matcher"));
+         *
+         * For more info see javadoc for Matchers class.
+         */
+        when(passwordEncoder.matches(any(), any()))
+                .thenReturn(true);
 
         // when
-        User result = userService.deleteById(ID);
+        User result = userService.login(request);
 
         // then
-        verify(userRepository).findById(ID);
-        verify(userRepository).delete(user);
         assertAll(
-                () -> assertEquals(user.getId(), result.getId()),
-                () -> assertEquals(user.getEmail(), result.getEmail()),
-                () -> assertEquals(user.getUsername(), result.getUsername()),
-                () -> assertEquals(user.getPassword(), result.getPassword()),
-                () -> assertEquals(user.getBio(), result.getBio()),
-                () -> assertEquals(user.getImage(), result.getImage())
+                () -> assertEquals(request.getEmail(), result.getEmail()),
+                () -> assertTrue(result.matchesPassword(request.getPassword(), passwordEncoder))
         );
     }
 
     @Test
-    void deleteById_exception() {
-
+    void save_validateDuplicateUser_exception() {
         // given
-        when(userRepository.findById(anyLong()))
-                .thenThrow(NoSuchElementException.class);
+        final UserJoinRequest request = UserJoinRequest.builder()
+                .email(EMAIL)
+                .username(USERNAME)
+                .password(PASSWORD)
+                .build();
+
+        final User user = User.builder()
+                .email(EMAIL)
+                .username(USERNAME)
+                .password(PASSWORD)
+                .bio(BIO)
+                .image(IMAGE)
+                .build();
+
+        when(userRepository.findByEmail(anyString()))
+                .thenReturn(Optional.ofNullable(user));
 
         // when
 
```

Comment:
```suggestion
        // when
        when(userRepository.findByEmail(anyString()))
                .thenReturn(Optional.ofNullable(user));
```

별건 아닌데 이게 맞는 순서인거 아닌가요? 아래 다른 코드들도 //when이 아래에 있네용